src/DataQI.Commons/Repository/Core/RepositoryFactory.cs

```diff
@@ -1,27 +1,33 @@
 using System;
+using DataQI.Commons.Util;
 
 namespace DataQI.Commons.Repository.Core
 {
     public abstract class RepositoryFactory
     {
-        public TRepository GetRepository<TRepository>()
+        public TRepository GetRepository<TRepository>(params object[] args)
             where TRepository : class
         {
-            var customImplementation = GetCustomImplementation(typeof(TRepository));
-            return GetRepository<TRepository>(customImplementation);
+            var repositoryInstance = GetRepositoryInstance(typeof(TRepository), args);
+            Assert.NotNull(repositoryInstance, "Repository Instance must not be null");
+
+            return GetRepository<TRepository>(() => repositoryInstance);
         }
 
-        public TRepository GetRepository<TRepository>(object customImplementation)
+        public TRepository GetRepository<TRepository>(Func<object> repositoryFactory)
             where TRepository : class
         {
-            return RepositoryProxy.Create<TRepository>(() => customImplementation);
+            Assert.NotNull(repositoryFactory, "Repository Factory must not be null");
```

Comment:
what do you think of using nameof method to reference repository factory variable?

Reply:
It can be, maybe I need to think how can I to communicate some problem a little better.